clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -898,6 +899,25 @@ static std::string convertCallArgsToString(Sema &S,
   return Result;
 }
 
+static std::string
+printCallArgsValueCategoryAndType(Sema &S, llvm::ArrayRef<const Expr *> Args) {
```

Comment:
I'm not really a fan of inventing printing mechanics outside of ASTPrinter but that probably integration header trauma is speaking. Let's see what the community says.